utils/decode_kallsyms.py

```diff
@@ -0,0 +1,45 @@
+import bisect
+import re
+
+
+def binary_search_recursive(array, element):
+    return bisect.bisect_right(array, element, key=lambda y: int(y.split()[0], 16)) - 1
+
+
+def cut_addresses_from_log_line_kernel_callstack(line):
+    my_regex = re.compile('0x([0-9a-f]{16})')
+    return my_regex.findall(line)
+
+
+def get_lines_from_log(file_path):
+    file_content = dict()
+    with open(file_path, 'r') as file:
+        for line in file:
+            if ('kernel callstack' in line and '0x' in line) or 'Reactor stalled for' in line:
+                file_content[line] = cut_addresses_from_log_line_kernel_callstack(line)
+    return file_content
+
+
+def get_kallsyms(path_to_kallsyms):
+    with open(path_to_kallsyms, 'r') as file:
+        file_content = file.readlines()
+    return file_content
+
+
+def append_content_to_file(file_path, content):
```

Comment:
the open with mode `w` is used only once, to clear the file from previous content, in the beginning of the function run, but we don't have a new to choose the mode like you suggest here, although now, we perhaps can only use open, since we are only passing the file stream to the writer function